rdt/transformers/id.py

```diff
@@ -164,27 +203,35 @@ def _fit(self, data):
         self.reset_randomization()
         self.data_length = len(data)
 
+        if self.cardinality_rule == 'match':
+            is_nan = int(pd.isna(data).any())  # nans count as a unique value
+            self._data_cardinality = data.nunique() + is_nan
+            self._unique_regex_values = self._generate_unique_regex_values()
```

Comment:
I believe this list will always have at least 1 value. It comes from the `strings_from_regex` method, which always produces a non-empty list, it either crashes for weird regexes like `(?!)` or `[1-0]{1}`, or it produces an empty string generator.

Throughout this file we make use of this assumption that this list doesn't have length 0.